internal/component/otelcol/receiver/cloudflare/cloudflare.go

```diff
@@ -0,0 +1,117 @@
+// Package cloudflare provides an otelcol.receiver.cloudflare component.
+package cloudflare
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/cloudflarereceiver"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pipeline"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/otelcol"
+	otelcolCfg "github.com/grafana/alloy/internal/component/otelcol/config"
+	"github.com/grafana/alloy/internal/component/otelcol/receiver"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/grafana/alloy/syntax"
+)
+
+var (
+	_ receiver.Arguments = Arguments{}
+	_ syntax.Defaulter   = (*Arguments)(nil)
+	_ syntax.Validator   = (*Arguments)(nil)
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "otelcol.receiver.cloudflare",
+		Stability: featuregate.StabilityExperimental,
+		Args:      Arguments{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := cloudflarereceiver.NewFactory()
+			return receiver.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+type Arguments struct {
+	Endpoint        string                      `alloy:"endpoint,attr"`
+	Secret          string                      `alloy:"secret,attr,optional"`
+	TimestampField  string                      `alloy:"timestamp_field,attr,optional"`
+	TimestampFormat string                      `alloy:"timestamp_format,attr,optional"`
+	Separator       string                      `alloy:"separator,attr,optional"`
+	Attributes      map[string]string           `alloy:"attributes,attr,optional"`
+	TLS             *otelcol.TLSServerArguments `alloy:"tls,block,optional"`
+
+	// Output configures where to send received data. Required.
+	Output *otelcol.ConsumerArguments `alloy:"output,block"`
+}
+
+// SetToDefault implements syntax.Defaulter.
+func (args *Arguments) SetToDefault() {
+	// Although otel's receiver already initializes defaults of downstream config,
+	// let's do it as well to avoid breaking changes if defaults are changed in upstream.
+	if args.TimestampField == "" {
+		args.TimestampField = "EdgeStartTimestamp"
+	}
```

Comment:
I could go either way on this/these. Generally, (since this is experimental stability and since the upstream dep is a 0.x release) even if OTel includes a breaking change, we will blindly carry that through to maintain the behavior that upstream has.

@dehaansa @ptodev any thoughts on this?

Reply:
@jharvey10 makes sense, then I'll let upstream to deal with defaults.

I have only one concern - component docs mention defaults for each attribute.\
We should track any update the docs every time upstream changes defaults.

Reply:
Definitely agree that we should track those changes. Typically when we upgrade otel each Alloy minor release, there's an exhaustive look through all of the changelogs and we pull out any deviations/breaking changes and call those out as needed in the changelog and docs). Our BAU process should cover what's needed for what you suggested to happen.

Reply:
@jharvey10 addressed in https://github.com/grafana/alloy/pull/4781/commits/c56e42e0902f85b9afe0fdd2404fbcebf5d947dc